tests/test_solar_gain.py

```diff
@@ -1,27 +1,18 @@
 from pythermalcomfort.models import solar_gain
-import json
-import requests
 
-# get file containing validation tables
-url = "https://raw.githubusercontent.com/FedericoTartarini/validation-data-comfort-models/main/validation_data.json"
-resp = requests.get(url)
-reference_tables = json.loads(resp.text)
 
-
-def test_solar_gain():
-    for table in reference_tables["reference_data"]["solar_gain"]:
-        for entry in table["data"]:
-            inputs = entry["inputs"]
-            outputs = entry["outputs"]
-            sg = solar_gain(
-                inputs["alt"],
-                inputs["sharp"],
-                inputs["I_dir"],
-                inputs["t_sol"],
-                inputs["f_svv"],
-                inputs["f_bes"],
-                inputs["asa"],
-                inputs["posture"],
-            )
-            assert sg["erf"] == outputs["erf"]
-            assert sg["delta_mrt"] == outputs["t_rsw"]
+def test_solar_gain(get_solar_gain_url, retrieve_data, is_equal):
+    reference_table = retrieve_data(get_solar_gain_url)
```

Comment:
there is a lot of code repetions, the function that extracts the:
* inputs
* outputs
* loops thought the keys and does the assertion 
are the same in all the test functions, we should create a function that basically does this automatically and we do not have to have code repetitions in all the tests